forge-gui/res/cardsfolder/upcoming/triumph_of_saint_katherine.txt

```diff
@@ -0,0 +1,11 @@
+Name:Triumph of Saint Katherine
+ManaCost:4 W
+Types:Creature Human Warrior
+PT:5/5
+K:Lifelink
+T:Mode$ ChangesZone | Origin$ Battlefield | OptionalDecider$ True | Destination$ Graveyard | ValidCard$ Card.Self | Execute$ TrigExile | TriggerDescription$ Praesidium Protectiva — When CARDNAME dies, exile it and the top six cards of your library in a face-down pile. If you do, shuffle that pile and put it back on top of your library.
+SVar:TrigExile:DB$ ChangeZone | Defined$ Self | Origin$ Battlefield | Destination$ Exile | RememberChanged$ True | SubAbility$ DBExileTopCard
+SVar:DBExileTopCard:DB$ Dig | Defined$ You | DigNum$ 6 | ChangeNum$ All | DestinationZone$ Exile | RememberChanged$ True | ExileFaceDown$ True | SubAbility$ DBChangeLibrary
+SVar:DBChangeLibrary:DB$ ChangeZone | Defined$ Remembered | Origin$ Exile | Destination$ Library | LibraryPosition$ 0 | RandomOrder$ True
```

Comment:
You need to handle this ruling:
> When Triumph of Saint Katherine's first ability resolves, it exiles as many of the relevant cards as possible. If you weren't able to exile all of those cards, which can happen if you have fewer than six cards in your library or if Triumph of Saint Katherine isn't in your graveyard as the ability resolves, the exiled cards will not be shuffled and returned to your library. They will remain exiled face-down.

Reply:
Any pointers to card I should look at ?

Reply:
I think you just need Condition on DBChangeLibrary if RememberedSize is 7

Reply:
GE6 means greater or equal
that's not 7